### PR TITLE
Add email field to personal info filter to identify coordinator from dashboard

### PR DESCRIPTION
**Pull request by dont-text-me**
~~Tiny PR to enable filtering of the personal info list by email - to be used in the Tooljet dashboard~~
This PR adds a dashboard viewset to retrieve personal info details - to be used to "authenticate" a coordinator upon first opening the dashboard app
image

### CODE REVIEW

File: tests/tests_api/test_personal_info.py [hunk 1]
```diff
@@ -29,20 +29,39 @@ def test_personal_info_create(api_client, fake_personal_info_data):
...
     )
     assert response.status_code == status.HTTP_200_OK
+    assert len(response.json()) == 1
+    # checking some basic attrs (cannot compare two objects directly
+    # because some fields were added during creation)
+    for param in ["first_name", "last_name", "email", "information_source"]:
+        assert response.json()[0][param] == fake_personal_info_data[param]
+
 
+def test_personal_info_get_with_params_returns_200_and_data_with_existing_email(
```
Comment (Auch-Auch) [inline tests/tests_api/test_personal_info.py:51]: The name is a little bit too big for my taste)

**Review by Auch-Auch — APPROVED** [review]

**Comment by lemontree210** [comment]
Thanks for the PR! Please extend the decorator so that `email` is shown in params in Swagger, like `registration_bot_chat_id` is.

File: api/views/personal_info.py [hunk 2]
```diff
@@ -56,6 +56,7 @@ def get_serializer_class(self) -> type[BaseSerializer[PersonalInfo]]:
     @extend_schema(
         parameters=[
             OpenApiParameter(name="registration_telegram_bot_chat_id", type=int, required=True),
+            OpenApiParameter(name="email", type=str, required=False),
```
Comment (lemontree210) [inline api/views/personal_info.py:59]:
```suggestion
```

Please revert this change, this is the wrong place for this parameter. It is a view for the bot that checks the existence of chat ID :)

**Review by lemontree210 — CHANGES_REQUESTED** [review]

**Comment by Auch-Auch** [comment]
> In general, we decided that for dashboard we use routes with `/api/dashboard/` and dashboard serializers. So we cannot just add `email` field to the viewset that is serving the bot. Also, we should use a dashboard serializer, not bot serializer. I should have more time and brain power to do it a bit later, maybe @Auch-Auch you can help out as you remember our logic of dividing viewsets and serializers.

Do you mean that we need a dashboard viewset here? ah, yep, I overlooked that, we should do the same as `students/teachers/coordinators`, so split it up into internal and dashboard. I try will to do it today or tomorrow evening if nobody does it earlier.

**Comment by dont-text-me** [comment]
ah, thanks for pointing that out! just managed to split things up, now looking like this: 
image

Not sure if there is any point in splitting up the filters too, the email field will just sit unused by the bot

**Review by lemontree210 — COMMENTED** [review]
Hm, I wonder if creation of a separate serializer class (`DashboardStandalonePersonalInfoSerializer`) is really warranted. I need @Roxe322 's expert opinion on that :)

**Review by Roxe322 — APPROVED** [review]

File: api/serializers/personal_info.py [hunk 3]
```diff
@@ -65,3 +69,18 @@ class Meta:
             "registration_telegram_bot_chat_id",
             "telegram_username",
         )
+
+
+class DashboardStandalonePersonalInfoSerializer(serializers.ModelSerializer[PersonalInfo]):
```
Comment (Roxe322) [inline api/serializers/personal_info.py:74]: Why "Standalone"? Maybe let's rename it to something like
`MinifiedDashboardPersonalInfoSerializer` as we have in other serializers?

File: tests/tests_api/test_personal_info.py [hunk 4]
```diff
@@ -29,26 +29,55 @@ def test_personal_info_create(api_client, fake_personal_info_data):
...
+    assert result["id"] == expected_id
+    assert result["email"] == expected_email
+
+
+def test_dashboard_personal_info_get_handles_unknown_email(api_client, fake_personal_info_list):
+    for item in fake_personal_info_list:
+        api_client.post("/api/personal_info/", data=item)
+
+    response = api_client.get(
+        path="/api/dashboard/personal_info/?email=unknownEmail@samanthasgroup.com",
```
Comment (Roxe322) [inline tests/tests_api/test_personal_info.py:75]:
```
api_client.get(
    "/api/dashboard/personal_info/",
    {"email": "unknownEmail@samanthasgroup.com"}
)
```

Comment (Roxe322) [reply tests/tests_api/test_personal_info.py:75]: same above

**Review by Roxe322 — CHANGES_REQUESTED** [review]

**Comment by Roxe322** [comment]
Totally agree with adding "minified" serializer with only needed fields @lemontree210

**Review by Roxe322 — APPROVED** [review]

**Comment by dont-text-me** [comment]
@lemontree210 i'll let you do the honours once youre happy with the pr👌

**Comment by lemontree210** [comment]
> @lemontree210 i'll let you do the honours once youre happy with the pr👌

Wait, what? 😆 

image

**Review by lemontree210 — APPROVED** [review]
the honours are mine :)

**Pull request merged**